docs/SNIPPET.txt

```diff
@@ -1 +1,17 @@
-TODO: add snippet
+main =
+    Path.fromStr "url.txt"
+    |> storeEmail
+    |> Task.onErr handleErr
+
+storeEmail = \path ->
+    url = File.readUtf8! path
+    user = Http.get! url Json.utf8
+    dest = Path.fromStr "$(user.name).txt"
+    File.writeUtf8! dest user.email
+    Stdout.line! "Wrote email to $(Path.display dest)"
+
+handleErr = \err ->
+    when err is
+        HttpErr url _ -> Stderr.line! "Error fetching URL $(url)"
+        FileReadErr path _ -> Stderr.line! "Error reading from $(Path.display path)"
+        FileWriteErr path _ -> Stderr.line! "Error writing to $(Path.display path)"
```

Comment:
This is a relatively long snippet.
Quoting the [docs](https://exercism.org/docs/building/tracks/docs#h-file-snippet-txt):

> The snippet should be relatively small (10-ish lines).

Reply:
Yeah, I just copied the code example from Roc's home page, but it normally comes with a nice interactive explanation for each line. I replaced this snippet with a simple `factorial` implementation. I included the `app [main] { pf: platform "https://..." }` to make sure the code snippet is actually runnable. The only problem with that is that it points to a specific version of the basic-cli platform (0.12.0), and we will have to update it regularly.